Week-6/Assignment-6/index.js

```diff
@@ -0,0 +1,81 @@
+const S1 = [-1, 2, 1, -4], T1 = 1;
+const S2 = [-3, 1, 2, 0, 6], T2 = 4;
+const S3 = [0, 0, 0], T3 = 2;
+const S4 = [-2], T4 = -2;
+const S5 = [4, 1], T5 = 4;
+
+/**
+ * @function validateInput checks if the inputs are valid
+ * @param {Array} arr an input array
+ * @param {Number} target the target sum
+ * @returns boolean
+ */
+const validateInput = (arr, target) => Array.isArray(arr) && arr.length > 2 &&
+    arr.every(el => Number.isFinite(el)) && Number.isInteger(target);
+
+/**
+ * @function getClosestSum a function which returns the sum closest to the target provided
+ * @summary Time Complexity - O(n^2), Space Complexity - O(1)
+ * @param {Array} arr an input array
+ * @param {Number} target the target sum
+ * @returns the sum closest to target
+ */
+const getClosestSum = (arr, target) => {
+    if (!validateInput(arr, target)) throw new Error('Invalid input provided');
+
+    const arrLen = arr.length;
+    let startIndex, endIndex = arrLen - 1;
+    let closestSum = Infinity;
+
+    // sorting the array first to have better time complexity
+    for (let i = 0; i < arrLen; i++) {
```

Comment:
Why can't you use inbuilt sorting mechanisms ?

Reply:
Thanks @sairamin, using arr.sort now